internal/storage/gcs/bucket.go

```diff
@@ -56,6 +86,34 @@ type Bucket interface {
 		ctx context.Context,
 		req *CreateObjectRequest) (*Object, error)
 
+	// Create a new object-writer according to the supplied request.
+	// This only returns an object-writer, to which content should be written.
+	// On closing this writer, a gcs.Object is returned.
+	//
+	// Sample usage:
+	// writer, err := bucket.CreateAsyncObjectWriter(ctx, createObjectReq,
+	//						chunkSize,
+	//						func(n int64) {
+	//							log("n bytes successfully uploaded")
+	//						}
+	// )
+	// // check err
+	// for n times {
+	// 		err = writer.Write(buffer)
+	// 		// check err
+	// }
+	// obj, err := writer.Close()
+	// // check err
+	//
+	// Official documentation:
+	//     https://cloud.google.com/storage/docs/json_api/v1/objects/insert
+	//     https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload
+	CreateAsyncObjectWriter(
```

Comment:
why is the method name asyncObjectWriter when it is returning objectWriter?

Reply:
Changed to CreateChunkUploader .